src/features/quiz/queries.ts

```diff
@@ -0,0 +1,64 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import quizzesApis from './apis';
+import { Quiz, QuizFilters } from './types';
+
+const quizKeys = {
+  all: ['quizzes'] as const,
+  lists: () => [...quizKeys.all, 'list'] as const,
+  list: (filters?: QuizFilters) => [...quizKeys.lists(), filters] as const,
+};
+
+const quizzesQueries = {
+  read: (params: QuizFilters) => {
+    return useQuery<Quiz[]>({
+      queryKey: quizKeys.lists(),
+      queryFn: () => quizzesApis.get(),
+      select: quizzes => {
+        const { partId, sectionId } = params;
+        return quizzes.filter(quiz => {
+          const matchesSection =
+            sectionId === 0 || quiz.sectionId === sectionId;
+          const matchesPart = partId === 0 || quiz.partId === partId;
+          return matchesSection && matchesPart;
+        });
+      },
+    });
+  },
+
+  create: () => {
+    const queryClient = useQueryClient();
+    return useMutation({
+      mutationFn: quizzesApis.post,
+      onSuccess: () => {
+        queryClient.invalidateQueries({
+          queryKey: quizKeys.lists(),
+        });
+      },
+    });
+  },
```

Comment:
```suggestion
  read: (params: QuizFilters) => {
    return useQuery<Quiz[]>({
      queryKey: quizKeys.lists(),
      queryFn: () => quizzesApis.get(),
      select: quizzes => {
        const { partId, sectionId } = params;
        return quizzes.filter(quiz => {
          const matchesSection =
            sectionId === 0 || quiz.sectionId === sectionId;
          const matchesPart = partId === 0 || quiz.partId === partId;
          return matchesSection && matchesPart;
        });
      },
    });
  },

  create: () => {
    const queryClient = useQueryClient();
    return useMutation({
      mutationFn: quizzesApis.post,
      onSuccess: () => {
        queryClient.invalidateQueries({
          queryKey: quizKeys.lists(),
        });
      },
    });
  },
```
ㅋㅋㅋ헉! 제가 약간의 오해를 드린 것 같습니다 ㅋㅋㅋㅋ 객체의 프로퍼티까지 줄바꿈으로 굳이 나누지는 않아도 됩니다 ㅎㅎㅎㅎㅎ

Reply:
아이고 ㅎㅎ;; 감사합니다!😢